### ISSUE TITLE
Finer grained span timing

### ISSUE THREAD

**Issue by brendanmaguire** [trace4cats/trace4cats#630]
Firstly, thanks for the great library :slightly_smiling_face: 

One thing I've noticed is that the spans are at millisecond granularity. Was this a deliberate decision? It seems like it would be useful to trace at microsecond granularity to facilitate fine grained tracing of function calls, etc.

**Comment by catostrophe** [comment]
@janstenpickle any objections?

**Comment by catostrophe** [comment]
CE3 has monotonic nano time

**Comment by brendanmaguire** [comment]
I've created a PR against CE3. https://github.com/typelevel/cats-effect/pull/2416

**Comment by catostrophe** [comment]
@brendanmaguire any ETA for the fix in CE3? When it's closed, do we need to fix anything in t4c?

**Comment by brendanmaguire** [comment]
@catostrophe The discussion seems to have stalled regarding adding that functionality. If you want to close this ticket in the meantime that's cool with me. I can reopen if there are updates on CE3.

**Comment by brendanmaguire** [comment]
The [cats effect PR](https://github.com/typelevel/cats-effect/pull/2416) has now been merged.

**Comment by catostrophe** [comment]
@brendanmaguire could you summarize what changes should be done in t4c in order to reflect those changes in CE3? Will it break anything in t4c?

**Comment by brendanmaguire** [comment]
t4c will get these changes for free once the change to CE3 has been released and t4c upgrades the version of the CE3 library it uses.

It won't break anyway unless t4c is relying on the fact that `IO.realTime` was previously returning the time at millisecond granularity and will now return it at microsecond granularity. I haven't seen anything in the code to indicate this will be a problem.

**Comment by catostrophe** [comment]
Ok. This issue probably may be closed, right?

**Comment by brendanmaguire** [comment]
Yes. I think we can close it.

**Comment by brendanmaguire** [comment]
For anyone interested in finer grained span timing, note that this will not be available until [Cats Effect 3.4.0](https://github.com/typelevel/cats-effect/milestone/20?closed=1) is released.
